Fixes for new mypy version - v0.990

`mypy` has changed its default behaviour so we need to apply some fixes / changes to the use of Optional in the code. This change has been introduced with `mypy==0.990`

```
openghg/plotting/_footprint.py:6: error: Incompatible default for argument "label" (default has type "None", argument has type "str")  [assignment]
openghg/plotting/_footprint.py:6: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
```

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -23,7 +23,7 @@ repos:
         exclude: doc
         exclude: tests
   - repo: https://github.com/pre-commit/mirrors-mypy
-    rev: v0.982
+    rev: v0.991
     hooks:
       - id: mypy
         # Copied from setup.cfg

--- a/openghg/dataobjects/_obsdata.py
+++ b/openghg/dataobjects/_obsdata.py
@@ -3,7 +3,7 @@ from dataclasses import dataclass
 from json import dumps
 from pathlib import Path
 from tempfile import NamedTemporaryFile
-from typing import Any, Dict, Iterator, Union
+from typing import Any, Dict, Iterator, Union, Optional
 
 import plotly.graph_objects as go
 
@@ -79,7 +79,11 @@ class ObsData(_BaseData, abc.Mapping):
         return to_transfer
 
     def plot_timeseries(
-        self, title: str = None, xlabel: str = None, ylabel: str = None, units: str = None
+        self,
+        title: Optional[str] = None,
+        xlabel: Optional[str] = None,
+        ylabel: Optional[str] = None,
+        units: Optional[str] = None,
     ) -> go.Figure:
         """Plot a timeseries"""
 

--- a/openghg/retrieve/_access.py
+++ b/openghg/retrieve/_access.py
@@ -19,7 +19,9 @@ logger = logging.getLogger("openghg.retrieve")
 logger.setLevel(logging.DEBUG)  # Have to set level for logger as well as handler
 
 DataTypes = Union[BoundaryConditionsData, FluxData, FootprintData, ObsColumnData, ObsData]
-multDataTypes = Union[List[BoundaryConditionsData], List[FluxData], List[FootprintData], List[ObsColumnData], List[ObsData]]
+multDataTypes = Union[
+    List[BoundaryConditionsData], List[FluxData], List[FootprintData], List[ObsColumnData], List[ObsData]
+]
 
 
 def _get_generic(
@@ -82,14 +84,14 @@ def _get_generic(
 def get_obs_surface(
     site: str,
     species: str,
-    inlet: str = None,
-    height: str = None,
-    start_date: Union[str, Timestamp] = None,
-    end_date: Union[str, Timestamp] = None,
-    average: str = None,
-    network: str = None,
-    instrument: str = None,
-    calibration_scale: str = None,
+    inlet: Optional[str] = None,
+    height: Optional[str] = None,
+    start_date: Optional[Union[str, Timestamp]] = None,
+    end_date: Optional[Union[str, Timestamp]] = None,
+    average: Optional[str] = None,
+    network: Optional[str] = None,
+    instrument: Optional[str] = None,
+    calibration_scale: Optional[str] = None,
     keep_missing: bool = False,
     skip_ranking: bool = False,
 ) -> Optional[ObsData]:
@@ -241,7 +243,7 @@ def get_obs_surface_local(
     # to be within the metadata (for now)
     if inlet is None and height is not None:
         inlet = height
-    inlet = format_inlet(inlet)    
+    inlet = format_inlet(inlet)
 
     site_info = load_json(filename="site_info.json")
     site = site.upper()
@@ -251,20 +253,21 @@ def get_obs_surface_local(
         raise ValueError(f"No site called {site}, please enter a valid site name.")
 
     surface_keywords = {
-        "site":site,
-        "species":species,
-        "inlet":inlet,
-        "start_date":start_date,
-        "end_date":end_date,
-        "network":network,
-        "instrument":instrument,
-        "data_type":data_type,
+        "site": site,
+        "species": species,
+        "inlet": inlet,
+        "start_date": start_date,
+        "end_date": end_date,
+        "network": network,
+        "instrument": instrument,
+        "data_type": data_type,
     }
 
     # # Get the observation data
     # obs_results = search_surface(**surface_keywords)
-    retrieved_data = _get_generic(ambig_check_params=["inlet", "network", "instrument"],
-                                  **surface_keywords)  # type:ignore
+    retrieved_data = _get_generic(
+        ambig_check_params=["inlet", "network", "instrument"], **surface_keywords  # type:ignore
+    )
 
     data = retrieved_data.data
 
@@ -273,7 +276,7 @@ def get_obs_surface_local(
         retrieved_data.metadata["inlet"] = "multiple"
 
     if start_date is not None and end_date is not None:
-        
+
         # Check if underlying data is timezone aware.
         data_time_index = data.indexes["time"]
         tzinfo = data_time_index.tzinfo
@@ -286,8 +289,8 @@ def get_obs_surface_local(
             end_date_filter = Timestamp(end_date)
 
         end_date_filter_exclusive = end_date_filter - Timedelta(
-                1, unit="nanosecond"
-            )  # Deduct 1 ns to make the end day (date) exclusive.
+            1, unit="nanosecond"
+        )  # Deduct 1 ns to make the end day (date) exclusive.
 
         # Slice the data to only cover the dates we're interested in
         data = data.sel(time=slice(start_date_filter, end_date_filter_exclusive))
@@ -558,12 +561,12 @@ def get_bc(
 def get_footprint(
     site: str,
     domain: str,
-    inlet: str = None,
-    height: str = None,
-    model: str = None,
-    start_date: Timestamp = None,
-    end_date: Timestamp = None,
-    species: str = None,
+    inlet: Optional[str] = None,
+    height: Optional[str] = None,
+    model: Optional[str] = None,
+    start_date: Optional[Timestamp] = None,
+    end_date: Optional[Timestamp] = None,
+    species: Optional[str] = None,
 ) -> FootprintData:
     """
     Get footprints from one site.
@@ -678,7 +681,7 @@ def _create_keyword_string(**kwargs: Any) -> str:
     This is used for printing details of keywords passed to the search functions.
     """
     used_keywords = {key: value for key, value in kwargs.items() if value is not None}
-    keyword_string = ', '.join([f"{key}='{value}'" for key, value in used_keywords.items()])
+    keyword_string = ", ".join([f"{key}='{value}'" for key, value in used_keywords.items()])
 
     return keyword_string
 
@@ -757,9 +760,9 @@ def _metadata_difference(
     return summary_difference
 
 
-def _metadata_difference_formatted(data: multDataTypes,
-                                   params: Optional[list] = None,
-                                   print_output: bool = True) -> str:
+def _metadata_difference_formatted(
+    data: multDataTypes, params: Optional[list] = None, print_output: bool = True
+) -> str:
     """
     Create formatted string for the difference in metadata between input objects.
 

--- a/openghg/standardise/surface/_npl.py
+++ b/openghg/standardise/surface/_npl.py
@@ -1,6 +1,6 @@
 from datetime import datetime
 from pathlib import Path
-from typing import Dict
+from typing import Dict, Optional
 
 from openghg.standardise.meta import assign_attributes
 from openghg.types import pathType
@@ -12,10 +12,10 @@ def parse_npl(
     data_filepath: pathType,
     site: str = "NPL",
     network: str = "LGHG",
-    inlet: str = None,
-    instrument: str = None,
-    sampling_period: str = None,
-    measurement_type: str = None,
+    inlet: Optional[str] = None,
+    instrument: Optional[str] = None,
+    sampling_period: Optional[str] = None,
+    measurement_type: Optional[str] = None,
 ) -> Dict:
     """Reads NPL data files and returns the UUIDS of the Datasources
     the processed data has been assigned to

--- a/openghg/store/_rank.py
+++ b/openghg/store/_rank.py
@@ -1,6 +1,6 @@
 from __future__ import annotations
 
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, Optional
 
 # from openghg.util import running_locally
 
@@ -8,7 +8,7 @@ if TYPE_CHECKING:
     from openghg.dataobjects import RankSources
 
 
-def rank_sources(site: str, species: str, service_url: str = None) -> RankSources:
+def rank_sources(site: str, species: str, service_url: Optional[str] = None) -> RankSources:
     """Retrieve datasources for a specific site and species. Returns a RankSources
     object that can be used to set and modify ranking data.
 

--- a/openghg/store/base/_datasource.py
+++ b/openghg/store/base/_datasource.py
@@ -586,9 +586,9 @@ class Datasource:
     @classmethod
     def load(
         cls: Type[T],
-        bucket: str = None,
-        uuid: str = None,
-        key: str = None,
+        bucket: Optional[str] = None,
+        uuid: Optional[str] = None,
+        key: Optional[str] = None,
         shallow: bool = False,
     ) -> T:
         """Load a Datasource from the object store either by name or UUID

--- a/requirements-dev.txt
+++ b/requirements-dev.txt
@@ -8,7 +8,7 @@ pytest-cov
 pytest-mock
 requests-mock
 # For mypy
-mypy==0.982
+mypy==0.991
 types-paramiko
 types-PyYAML
 types-requests

